LCM/dsc/engine/dsc_host/dsc_host.c

```diff
@@ -0,0 +1,324 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <locale.h>
+
+#include "dsc_sal.h"
+#include "DSC_Systemcalls.h"
+#include "Resources_LCM.h"
+#include "EngineHelper.h"
+#include "dsc_host.h"
+#include "dsc_library.h"
+#include "lcm/strings.h"
+
+MI_Result GetDSCHostVersion(_In_z_ MI_Char* version, _In_ size_t length)
+{
+    if(Stprintf(version, length, MI_T("Version : %d.%d.%04d"), MajorVersion, MinorVersion, BuildVersion) < 0)
+        return MI_RESULT_FAILED;
+    else
+        return MI_RESULT_OK;
+}
+
+void PrintVersion()
+{
+    MI_Char versionInfo[DSCHOST_VERSION_BUFF_SIZE];
+    GetDSCHostVersion(versionInfo, DSCHOST_VERSION_BUFF_SIZE);
+    Tprintf(versionInfo);
+}
+
+void PrintHelp()
+{
+    Tprintf(MI_T("Usage:\n"));
+    Tprintf(MI_T("dsc_host [--help] [--version]\n"));
+    Tprintf(MI_T("dsc_host <Output Folder Path> <Operation> [Operation Arguments] \n"));
+    Tprintf(MI_T("\n"));
+    Tprintf(MI_T("Supported Operation values are:\n"));
+    Tprintf(MI_T("  SendConfiguration <MOF Document Path>\n"));
+    Tprintf(MI_T("  SendConfigurationApply <MOF Document Path>\n"));
+    Tprintf(MI_T("  GetConfiguration [Configuration Document Path]\n"));
+    Tprintf(MI_T("  ApplyConfiguration\n"));
+    Tprintf(MI_T("  SendMetaConfigurationApply <MOF Document Path> [force]\n"));
+    Tprintf(MI_T("  GetMetaConfiguration\n"));
+    Tprintf(MI_T("  RollBack\n"));
+    Tprintf(MI_T("  TestConfiguration\n"));
+    Tprintf(MI_T("  PerformRequiredConfigurationChecks [flags value]\n"));
+    Tprintf(MI_T("  StopConfiguration [force]\n"));
+    Tprintf(MI_T("  PerformInventory\n"));
+    Tprintf(MI_T("  PerformInventoryOOB [MOF Document Path]\n"));
+    Tprintf(MI_T("\n"));
+    Tprintf(MI_T("Examples:\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendConfiguration ./new_config.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendConfigurationApply ./new_config.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output GetConfiguration ./GetAuditPolicy.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output ApplyConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output SendMetaConfigurationApply ./new_metaconfig.mof\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output GetMetaConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output RollBack\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output TestConfiguration\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output PerformRequiredConfigurationChecks 1\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output StopConfiguration force\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output PerformInventory\n"));
+    Tprintf(MI_T("  /opt/dsc/bin/dsc_host /opt/dsc/output PerformInventoryOOB ./Inventory.mof\n"));
+    Tprintf(MI_T("\n"));
+}
+
+int main(int argc, char *argv[])
```

Comment:
>ain [](start = 5, length = 3)

is crone jobs change coming on separate PR.  who is calling this .exe

Reply:
No need to change the cron jobs :) That part stays as is.

---
In reply to: [263592511](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/541#discussion_r263592511) [](ancestors = 263592511)

Reply:
Cron job or user call our python scripts which in turn call this binary. Design document has more details on this.

---
In reply to: [264081728](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/541#discussion_r264081728) [](ancestors = 264081728,263592511)